src/jit/gentree.cpp

```diff
@@ -15158,11 +15158,12 @@ Compiler::TypeProducerKind Compiler::gtGetTypeProducerKind(GenTree* tree)
     }
     else
     {
-        bool                 isExact   = false;
-        bool                 isNonNull = false;
-        CORINFO_CLASS_HANDLE clsHnd    = gtGetClassHandle(tree, &isExact, &isNonNull);
+        bool                 isExact        = false;
+        bool                 isNonNull      = false;
+        CORINFO_CLASS_HANDLE clsHnd         = gtGetClassHandle(tree, &isExact, &isNonNull);
+        CORINFO_CLASS_HANDLE runtimeTypeHnd = info.compCompHnd->getBuiltinClass(CLASSID_RUNTIME_TYPE);
 
-        if (clsHnd == info.compCompHnd->getBuiltinClass(CLASSID_RUNTIME_TYPE))
+        if (runtimeTypeHnd != (CORINFO_CLASS_HANDLE)nullptr && clsHnd == runtimeTypeHnd)
```

Comment:
Would it be better to do this instead?
```
if (clsHnd != NULL && clsHnd == info.compCompHnd->getBuiltinClass(CLASSID_RUNTIME_TYPE))
```

It saves the call to getBuiltinClass when clsHnd is null.

Reply:
Yes it would, thanks!